### ISSUE TITLE
No circularity error for self referential get accessor annotations

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#48031]
```ts
declare class C {
    // TypeScript correctly reports an error here.
    foo: typeof this.foo;
    // TypeScript does not report an error here.
    get bar(): typeof this.bar;
}
```

**Expected**

An error on both `foo` and `bar`.

```
'foo' is referenced directly or indirectly in its own type annotation.
'bar' is referenced directly or indirectly in its own type annotation.
```

**Actual**

Just an error on `bar`.

```
'bar' is referenced directly or indirectly in its own type annotation.
```

**Comment by fatcerberus** [comment]
The actual error(s) I'm seeing are rather weirder than this issue would suggest.

> * 'foo' is referenced directly or indirectly in its own type annotation.
> * 'bar' implicitly has return type 'any' because it does not have a return type annotation and is referenced directly or indirectly in one of its return expressions.
> * Public property 'foo' of exported class has or is using private name 'this'.
> * Return type of public getter 'bar' from exported class has or is using private name 'this'.

[Playground](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEkoGdnwMLwN4Ch7-gHpD4AVATwAcQBlMGAS0oBdEB7GOMZic+OSh2ZooAO3ghOHeAAtJIAHR4CAMzZsAXPGZUQbFdpkNkCtWwDcy-MTK66jFvGBsQaUW1YCh8MRKkxZeSUCeABzEFYAI1gACgBKLR1qfUNjBWiYSwBfHCA)

**Comment by fatcerberus** [comment]
I don't see the bug?  The error on `bar` is slightly different from the one on `foo`, but both have self-referential type errors.
